src/plans/D3MCompoundPlan.sol

```diff
@@ -0,0 +1,187 @@
+        if (what == "barb") {
+            barb = data;
+        } else revert("D3MCompoundPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+    function file(bytes32 what, address data) external auth {
+        if (what == "tack") tack = InterestRateModelLike(data);
+        else revert("D3MCompoundPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function getTargetAssets(uint256 currentAssets) external override view returns (uint256) {
+        uint256 targetInterestRate = barb;
+        if (targetInterestRate == 0) return 0; // De-activated
+
+        uint256 borrows = cDai.totalBorrows();
+        uint256 targetTotalPoolSize = _calculateTargetSupply(targetInterestRate, borrows);
+        uint256 totalPoolSize = cDai.getCash() + borrows - cDai.totalReserves();
+
+        if (targetTotalPoolSize >= totalPoolSize) {
+            // Increase debt (or same)
+            return currentAssets + targetTotalPoolSize - totalPoolSize;
+        } else {
+            // Decrease debt
+            unchecked {
+                uint256 decrease = totalPoolSize - targetTotalPoolSize;
+                if (currentAssets >= decrease) {
+                    return currentAssets - decrease;
+                } else {
+                    return 0;
+                }
+            }
+        }
+    }
+
+    function _calculateTargetSupply(uint256 targetInterestRate, uint256 borrows) internal view returns (uint256) {
+        uint256 kink                   = tack.kink();
+        uint256 multiplierPerBlock     = tack.multiplierPerBlock();
+        uint256 baseRatePerBlock       = tack.baseRatePerBlock();
+        uint256 jumpMultiplierPerBlock = tack.jumpMultiplierPerBlock();
+
+        uint256 normalRate = _wmul(kink, multiplierPerBlock) + baseRatePerBlock;
+
+        uint256 targetUtil;
+        if (targetInterestRate > normalRate) {
+            if (jumpMultiplierPerBlock == 0) return 0; // illegal rate, max is normal rate for this case
+            targetUtil = kink + _wdiv(targetInterestRate - normalRate, jumpMultiplierPerBlock); // (1)
+        } else if (targetInterestRate > baseRatePerBlock) {
+            targetUtil = _wdiv(targetInterestRate - baseRatePerBlock, multiplierPerBlock);      // (2)
+        } else {
+            // if (target == base) => (borrows == 0) => supply does not matter
+            // if (target  < base) => illegal rate
+            return 0;
+        }
+
+        if (targetUtil > WAD) return 0; // illegal rate (unacheivable utilization)
+
+        return _wdiv(borrows, targetUtil);                                                      // (3)
+    }
+
+    function active() public view override returns (bool) {
+        if (barb == 0) return false;
+        return CErc20Like(cDai).interestRateModel() == address(tack);
+    }
+
+    function disable() external override {
+        require(wards[msg.sender] == 1 || !active(), "D3MCompoundPlan/not-authorized");
```

Comment:
What's the reasoning behind allowing anyone to disable if the there's an interest rate model mismatch? I don't really see an issue with it right now, but just wondering if it's actually _necessary_ for some reason?

Reply:
Just discussed it with the others. The idea was that if the rate model on Aave/Compound changes anyone can permanently disable the module, so that even ithe model on Aave/Compound returns to the original one it will still be disabled.
Motivation was to not support this kind of back and forth model changing.

Reply:
Okay but why don't we want to support re-enabling if the other protocol returns to a trusted plan? Just to force governance to re-evaluate? Also do we care if no one bothers to call `disable` in such a situation?

Reply:
Yeah this has been discussed in the V2 reviews, the idea is to force governance to re-evaluate.
We indeed need someone to call `disable` for not supporting returning to an old supported model. It was decided we prefer that then the alternative of making `active` a non view function.